code/search/FullTextSearch.php

```diff
@@ -18,16 +18,41 @@ class FullTextSearch {
 	 * @param bool $rebuild - If true, don't use cached values
 	 */
 	static function get_indexes($class = null, $rebuild = false) {
-		if ($rebuild) { self::$all_indexes = null; self::$indexes_by_subclass = array(); }
+		if ($rebuild) {
+			self::$all_indexes = null;
+			self::$indexes_by_subclass = array();
+		}
 
 		if (!$class) {
 			if (self::$all_indexes === null) {
 				$classes = ClassInfo::subclassesFor('SearchIndex');
 
-				$concrete = array();
+				$hidden = array();
+				$candidates = array();
 				foreach ($classes as $class) {
+					// Check if this index is disabled
+					$hides = $class::config()->hide_ancestor;
```

Comment:
I wanted to go for something that, if ignored, doesn't have any effect, and is fully backwards compatible.

You can do the same in yaml though.

```
MyNewIndex:
  hide_ancestor: MyOldIndex
```

Ok, maybe I can split it into a new PR.